admission_control/admission_control_service/src/runtime.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{admission_control_service::AdmissionControlService, upstream_proxy::UpstreamProxy};
+use admission_control_proto::proto::admission_control_grpc::{
+    create_admission_control, AdmissionControlClient,
+};
+use config::config::NodeConfig;
+use futures::{
+    channel::mpsc,
+    future::{FutureExt, TryFutureExt},
+};
+use grpc_helpers::ServerHandle;
+use grpcio::{ChannelBuilder, EnvBuilder, ServerBuilder};
+use grpcio_sys;
+use mempool::proto::mempool_grpc::MempoolClient;
+use network::{
+    proto::AdmissionControlMsg,
+    validator_network::{AdmissionControlNetworkEvents, AdmissionControlNetworkSender},
+};
+use std::{cmp::min, sync::Arc};
+use storage_client::{StorageRead, StorageReadServiceClient};
+use tokio::runtime::{Builder, Runtime};
+use vm_validator::vm_validator::VMValidator;
+
+/// Handle for AdmissionControl Runtime
+pub struct AdmissionControlRuntime {
+    /// gRPC server to serve request between client and AC
+    pub grpc_server: ServerHandle,
+    /// separate shared mempool runtime
+    pub upstream_proxy: Runtime,
+    /// AC Client
+    pub client: AdmissionControlClient,
+    /// true if validator, false if full node
+    pub is_validator: bool,
```

Comment:
why do we need to store it? can't we just check config on a fly?

Reply:
Because the check has been done in main_node.rs already, so thought it would be better to just pass it in